### PR TITLE
Support 32-bit architecture

### LINKED ISSUE #55: `architecture` parameter is ignored
I'm trying to install an x86 JDK in a GitHub action, and I found out that the specified architecture is ignored in this script. By looking at the code, I saw that `arch` is not used when the script is determining which archive to download. See https://github.com/actions/setup-java/blob/5c87b70ffefc9ca2e77153b3528c6b7bc85a84a9/src/installer.ts#L194-L203 On Windows, it will always download the archive ending with `-win_x64`, while it should actually download the archive ending with `-win_i686.zip` when the architecture is `x86`. The issue also seems to be present for other OSes (although I never tried them)

Be aware that Zulu doesn't seem to provide x86 Windows binaries for all Java versions.

### PR DESCRIPTION
**Pull request by emgre**
Fix #55 

I'm using this code in this GitHub Actions pipeline: https://github.com/dnp3/opendnp3/actions

I only tried it on Windows. Perhaps some integration tests could be added to verify that the proper version is fetched. I'm not sure to understand how your tests are run, so I didn't do it.

### CODE REVIEW

File: src/installer.ts [hunk 1]
```diff
@@ -188,20 +188,33 @@ async function unzipJavaDownload(
...
   refs: string[],
   version: string,
+  arch: string,
   javaPackage: string
 ): {version: string; url: string} {
   version = normalizeVersion(version);
+
+  let archExtension = '';
+  if (arch === 'x86') {
+    archExtension = 'i686';
```
Comment (joshmgross) [inline src/installer.ts:198]: Where did this extension come from? Is it not `x86`?

File: src/installer.ts [hunk 2]
```diff
@@ -188,20 +188,33 @@ async function unzipJavaDownload(
...
+  } else if (arch === 'x64') {
+    archExtension = 'x64';
+  } else {
+    throw new Error(`architecture "${arch}" is not int [x86 | x64]`);
```
Comment (joshmgross) [inline src/installer.ts:202]: We should validate that the architecture input is valid when we retrieve the input in https://github.com/actions/setup-java/blob/master/src/setup-java.ts#L12

Then this whole block can just be:
```
const archExtension = arch === 'x86' ? 'i686' : 'x64';
```

File: src/installer.ts (hunk 1 above)
Comment (emgre) [reply src/installer.ts:198]: `i686` is the latest iteration of the `IA-32` (commonly known as `x86`). From today's perspective, it's a synonym of `x86` (although it is technically not). I didn't choose this extension, I just looked at https://cdn.azul.com/zulu/bin/ and saw that's what they use.

File: src/installer.ts (hunk 2 above)
Comment (emgre) [reply src/installer.ts:202]: Indeed. I'm not very familiar with the GitHub Actions core library (and JS in general) so bear with me. I'll make the change soon.

Comment (joshmgross) [reply src/installer.ts:202]: No worries! Let me know if you need any help

**Comment by AustinShalit** [comment]
Hi folks. Any progress updates on this PR?

File: src/installer.ts (hunk 2 above)
Comment (PeterJohnson) [reply src/installer.ts:202]: Should be "not in" instead of "not int".

**Comment by AustinShalit** [comment]
@konradpabjan This PR can now be closed.

**Comment by joshmgross** [comment]
Replaced with https://github.com/actions/setup-java/pull/95

**Pull request closed without merging**